### ISSUE TITLE
android version making unwanted automatic connections

### ISSUE THREAD

**Issue by ureru** [Eloston/ungoogled-chromium#1297]
It's quite simple, the android version (I haven't tried the PC version) of ungoogled-chromium is making unwanted automatic connections, and not just one, it's several different IP addresses. Several of them are google, and there are also some for app-measurement.com. The rest appear to be github, those are at least less concerning but surely not what the project is intended to do.

google:
64.233.177.95
64.233.177.155
64.233.177.157
64.233.185.95
74.125.21.156
74.125.138.95
108.177.122.95
108.177.122.102
108.177.122.113
108.177.122.139
142.250.9.155
142.250.9.156
173.194.219.95

app-measurement.com:
74.125.21.101
74.125.21.113

github:
3.94.147.15
140.82.114.3
140.82.114.4
140.82.114.5
151.101.204.133
185.199.109.154
199.232.32.133

Steps to reproduce:
1. Launch ungoogled-chromium while using some software to view a traffic log for your device. Whether you visit web sites or not you will see unwanted connections pop up automatically.

Expected behavior:
None of these automatic connections occur.

 - OS/Platform and version: Android 10
 - ungoogled-chromium version: 86.0.4240.111

**Comment by wchen342** [comment]
Can you describe in detail how you tested the connections? I just double checked and there is certainly no mention of anything related to `app-measurement.com` and `github.com` in the chromium source code, and `google.com` are supposedly blocked by domain substitution.

If you monitored the connections outside your Android device it is possible the connections come from other processes on you system.

**Comment by ureru** [comment]
The monitoring is taking place on the device.
1. Download TrackerControl off of f-droid.
2. Tap the 3 dots in the top right.
3. Tap "Traffic log."
4. Tap the little slider thing at the top middle to enable the traffic log.
5. The three lines at the top right allow you to configure what kinds of things are shown in the log. The three dots at the top right allow you to configure some things that include whether or not it should show you more information than just the IPs.

**Comment by wchen342** [comment]
I tried on my own machine but cannot make it happen. Did you try to visit Google sites when testing? UC only makes sure the browser itself doesn't make connections to google server, but it won't block the trackers if the website you visit has them.

![screenshot-2020-12-25_02 37 19 893](https://user-images.githubusercontent.com/6406743/103111978-d46f5880-465a-11eb-8769-5713459a7ffe.png)

**Comment by ureru** [comment]
Did you tap the 3 lines in the top right that I mentioned and select "Blocked"? If you didn't, it won't show you the connections that TrackerControl has automatically blocked according to the Disconnect.me blocklist. I suspect this is what's going on as I've already seen on my device that TrackerControl blocks these connections that I reported.

**Comment by wchen342** [comment]
Yeah it is selected by default when installed:

![screenshot-2020-12-25_05 43 37 698](https://user-images.githubusercontent.com/6406743/103117708-5f108180-4674-11eb-95a2-4b377aa0f74a.png)

**Comment by ureru** [comment]
Well what am I supposed to make of that? Did I get a wrong build of it maybe? I got it from the f-droid repository. Do you have any idea what's going on?

**Comment by wchen342** [comment]
I am not sure at this point. Can you try the new v87 version [here](https://github.com/ungoogled-software/ungoogled-chromium-android/releases/tag/87.0.4280.88-1)?

**Comment by wchen342** [comment]
I think I figured out where the Github connections come from. It is the link on new tab and chromium tries to fetch its content.

It seems the Google connections are not made on `x86` (or `debug`) version for some reason. But I now see connections on release version. I will investigate this further.

**Comment by wchen342** [comment]
Actually it happens when I clear all data and return to the new tab page, and chromium starts to connect to `safebrowsing.googleapis.com` which is very strange because safebrowsing is supposedly disabled completely. What is even stranger is that I cannot find any reference to the address in any codes.
@csagan5 Do you have any idea about this?

**Comment by csagan5** [comment]
You are using different patches here, so it cannot really be compared to Bromite without comparing those patches first. The issue does not seem to happen on Bromite.

You need to test with a fresh install in any case (having open tabs taints the test); look in the code for `kSbV4UrlPrefix`: is it replaced? I would say that the safe browsing functionality is not completely disabled.

**Comment by wchen342** [comment]
> look in the code for kSbV4UrlPrefix

The variable is replaced by domain substitution so I wonder where else can this address come from. I used `grep -rI 'safebrowsing.googleapis'` in the `src` folder but cannot find anything. That is what puzzles me.

> I would say that the safe browsing functionality is not completely disabled.

Is there anything you know that will still be functioning when built with `safe_browsing_mode=0`?

**Comment by csagan5** [comment]
`safe_browsing_mode=0` should disable everything; my best advice is to inspect the actual requests' content as it was done [here](https://github.com/bromite/bromite/issues/471#issuecomment-577982232) by @SnwMds ~1 year ago for Bromite.

You might find out more clues about what exactly is being sent to that server.

See also:
* https://www.chromium.org/developers/design-documents/safebrowsing
* https://web.archive.org/web/20171204063618/https://developers.google.com/safe-browsing/v3/
* https://developers.google.com/safe-browsing/v4/

**Comment by ureru** [comment]
I tried the new uc 87 and had the same issue. It wasn't making any connections besides a single github connection for less than a minute, at which point it started doing this. After the screenshot it happened around another 46 times over time. This is all before loading a single web page. UC was only showing the new tab page.
https://f.secretalgorithm.com/UASJs/ss1.png

**Comment by wchen342** [comment]
I got the google apis calls. It was exactly the same calls in the Bromite issue. Haven't been able to reproduce the safebrowsing calls though.

I will import the patch disabling `MediaDrmPreprovisioning` for next version.

**Comment by wchen342** [comment]
Shall be solved in [9b95e90](https://github.com/ungoogled-software/ungoogled-chromium-android/commit/9b95e90791010e93971877cf4baf699afa69a663).

**Comment by csagan5** [comment]
DRM provisioning is unrelated to safe browsing.

**Comment by wchen342** [comment]
I didn't explain it in detail above but there are two kinds of connections I saw, one is to `www.googleapis.com` which is what the original issue is describing. The safebrowing connection was only seen once when I was debugging and I asked above because I thought they are probably related, but I cannot reproduce the connection to `safebrowing.googleapis.com` since then so I cannot debug that one. The one I catched is the connection to `www.googleapis.com` which is from DRM preprovisioning and so is fixed with that patch.
